subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishDependenciesIntegTest.groovy

```diff
@@ -158,11 +158,15 @@ class MavenPublishDependenciesIntegTest extends AbstractIntegrationSpec {
 
         where:
         plugin         | gradleConfiguration  | mavenScope
-        'java'         | 'compile'            | 'runtime'
+        'java'         | 'compile'            | 'compile'
+        'java'         | 'runtime'            | 'compile'
```

Comment:
`runtime` is a deprecated scope which has always been leaking into the compile classpath of downstream projects. The better replacement is `runtimeOnly`, which does not have this leakage. This change just straightens out the behavior so that Maven publishing behaves exactly like a multi-project build would behave, i.e. leaking those dependencies into the compile classpath of consumers.